src/app/admin/services/backup.service.ts

```diff
@@ -15,35 +15,36 @@ export class BackupService {
 
   getJsonExport(): Promise<string> {
     return this.db.getAll()
-      .then(results => {
-        let res = '';
-        results.forEach(r => {
-          res += JSON.stringify(r) + BackupService.SEPARATOR_ROW;
-        });
+      .then(results => this.createJson(results));
+  }
 
-        return res.trim();
-      });
+  createJson(data): string {
+    let res = '';
+    data.forEach(r => {
+      res += JSON.stringify(r) + BackupService.SEPARATOR_ROW;
+    });
+
+    return res.trim();
   }
 
   getCsvExport(): Promise<string> {
     return this.db.getAll()
-      .then(results => {
-        const resultFields = ['_id', '_rev'];
-        results.forEach(r => {
-          for (const propertyName in r) {
-            if (resultFields.indexOf(propertyName) === -1) {
-              resultFields.push(propertyName);
-            }
-          }
-        });
-
-        return this.papa.unparse(
-          {data: results, fields: resultFields},
-          {quotes: true, header: true});
-      });
+      .then(results => this.createCsv(results));
   }
 
-
+  createCsv(data): string {
+    const resultFields = ['_id', '_rev'];
+    data.forEach(d => {
```

Comment:
I find it a bit hard to understand what this does, but am also unsure how to simplify. A really minor thing is to replace `indexOf` with `includes`. Also having mixed `.forEach()` and `for ... in` feels a bit strange.

All in all, the backup service could benefit from using async functions instead of using Promises directly, but that's just my personal opinion.

Reply:
I don't think the async await syntax would help much here. There is barely more then one asynchronous call per funtion.